cpp/src/arrow/flight/sql/odbc/flight_sql/get_info_cache.cc

```diff
@@ -1332,6 +1336,7 @@ void GetInfoCache::LoadDefaultsForMissingEntries() {
                           SQL_FN_TSI_FRAC_SECOND | SQL_FN_TSI_SECOND | SQL_FN_TSI_MINUTE |
                           SQL_FN_TSI_HOUR | SQL_FN_TSI_DAY | SQL_FN_TSI_WEEK |
                           SQL_FN_TSI_MONTH | SQL_FN_TSI_QUARTER | SQL_FN_TSI_YEAR));
+  SetDefaultIfMissing(info_, SQL_TIMEDATE_FUNCTIONS, static_cast<uint32_t>(0));
```

Comment:
For `SQL_TIMEDATE_FUNCTIONS`,  we can return bit masks to indicate the functionality is supported.